pkg/clients/database/mysql.go

```diff
@@ -133,26 +139,11 @@ func (c *MySQLServerClient) CreateServer(ctx context.Context, cr *azuredbv1beta1
 	if err != nil {
 		return err
 	}
-	// NOTE(muvaf): There are cases where Create call does not return error
-	// and you don't see any SQL instance in the console UI. In those cases,
-	// since name is not taken, we always call Create with a new password each time.
-	// The problem is, error that blocks creation happens after the Create call
-	// is initiated.
-	// DoneWithContext checks the operation once and errors usually surface
-	// themselves in the first check, if there is one we are able to warn the user
-	// for a fix.
-	// However, there could be cases where the error appears after a long while.
-	// For that reason, some libraries make use of WithCompletionRef function. However,
-	// that function blocks until success or error with the context deadline.
-	// Azure does not provide any guarantees regarding provisioning time and it
-	// can get really long. In most of the cases, our context deadline gets reached.
-	//
-	// For the time being, we check only once to cover the greater possibility of
-	// capturing the error. If there is none, we'll never call Create again anyway.
-	// If the problem happens after a while, we'd fallback to old behavior where
-	// we continuously try to create and fail.
-	_, err = op.DoneWithContext(ctx, c.ServersClient.Client)
-	return err
+	cr.Status.AtProvider.LastOperation = v1alpha3.AsyncOperation{
+		PollingURL: op.PollingURL(),
+		Method:     http.MethodPut,
+	}
+	return azure.FetchAsyncOperation(ctx, c.ServersClient.Client, &cr.Status.AtProvider.LastOperation)
```

Comment:
Would it make sense to have these client level CRUD abstractions return some summary of of the operation, rather than updating the `*azuredbv1beta1.MySQLServer` they're passed? My thinking is:

* We're currently split between calling `azure.FetchAsyncOperation` inside the client and outside of the client (in `Observe`). It could be nice to be consistent and always update the operation from the `ExternalClient`.
* I _think_ most of our "client" types in Crossplane are read-only; they translate the Crossplane type to the cloud provider type and submit it. This makes me feel a (tiny) bit uncomfortable about these methods updating the type they're passed.

Reply:
> It could be nice to be consistent and always update the operation from the ExternalClient.

Yes, makes sense. Moved the fetch calls to `ExternalClient`. IMO, actual `Create/Update/Delete/Get` calls should also be called in `ExternalClient` with the converted objects from client package but that's probably for another PR.